hwloc/topology.c

```diff
@@ -1586,7 +1613,12 @@ propagate_unused_cpuset(hwloc_obj_t obj, hwloc_obj_t sys)
 	hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, sys->complete_cpuset);
       } else {
 	obj->complete_cpuset = hwloc_bitmap_dup(sys->complete_cpuset);
-	hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, obj->cpuset);
+        if (obj->complete_cpuset)
+	  hwloc_bitmap_and(obj->complete_cpuset, obj->complete_cpuset, obj->cpuset);
+        else {
+          hwloc_bitmap_free(mask);
+          return -1;
+        }
       }
```

Comment:
While code is being reworked in task branch we still continue to use master and if I remove it our code may crash. Or I misunderstood You?